repos/system_upgrade/common/actors/peseventsscanner/libraries/peseventsscanner_repomap.py

```diff
@@ -0,0 +1,296 @@
+                pesid_repos.append(pesid_repo)
+        return pesid_repos
+
+    def get_source_pesid_repos(self, pesid):
+        """
+        Return the list of PESIDRepositoryEntry objects for a specified PES ID
+        mathing the source OS major version.
+
+        :param pesid: The PES ID for which to retrieve PESIDRepositoryEntries.
+        :type pesid: str
+        :return: A list of PESIDRepositoryEntries that match the provided PES ID and have
+                 the OS Major version same as the source OS.
+        :rtype: List[PESIDRepositoryEntry]
+        """
+        return self.get_pesid_repos(pesid, get_source_major_version())
+
+    def get_target_pesid_repos(self, pesid):
+        """
+        Return the list of PESIDRepositoryEntry objects for a specified PES ID
+        mathing the target OS major version.
+
+        :param pesid: The PES ID for which to retrieve PESIDRepositoryEntries.
+        :type pesid: str
+        :return: A list of PESIDRepositoryEntries that match the provided PES ID and have
+                 the OS Major version same as the target OS.
+        :rtype: List[PESIDRepositoryEntry]
+        """
+        return self.get_pesid_repos(pesid, get_target_major_version())
+
+    def _find_repository_target_equivalent(self, src_pesidrepo, target_pesid):
+        """
+        Find the target repository that is the best-match to the source one with the given
+        target PES ID.
+
+        :param src_pesidrepo: The source repository to find equivalent to.
+        :type src_pesidrepo: PESIDRepositoryEntry
+        :param target_pesid: The target PES ID which the target repository must contain.
+        :type target_pesid: str
+        :return: A target equivalent of given repository.
+        :rtype: Optional[PESIDRepositoryEntry]
+        """
+
+        candidates = []
+        for candidate in self.get_target_pesid_repos(target_pesid):
+            matches_rhui = candidate.rhui == src_pesidrepo.rhui
+            matches_repo_type = candidate.repo_type == 'rpm'
+            matches_arch = candidate.arch == api.current_actor().configuration.architecture
+
+            if matches_rhui and matches_arch and matches_repo_type:
+                # user can specify in future the specific channel should be
+                # prioritized always (e.g. want to go to EUS...).
+                channel = self.prio_channel or src_pesidrepo.channel
+                if candidate.channel == channel:
+                    return candidate
+                candidates.append(candidate)
+
+        # Fallback...
+        # Could not find exact-match, so go through candidates if we find an
+        # alternative in one of default channels (usually just 'ga')
+        for channel in self.default_channels:
+            for candidate in candidates:
+                if channel == candidate.channel:
+                    return candidate
+
+        # This is ...weird case (handling out of the method)
```

Comment:
Let's be more specific here, you mean the case when there is no single repository enabled on the target system?

Reply:
no, when the equivalent target repository has not ben discovered (which is what the method is documented to deliver). It's weird case we do not expect it should happen but possibly it could. for completeness it's there. However more likely it points to incomplete data or a business decision leading to such a data.